src/main/java/org/broadinstitute/hellbender/tools/GetSampleName.java

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.hellbender.tools;
+
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.BetaFeature;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.QCProgramGroup;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@CommandLineProgramProperties(
+        oneLineSummary = "(EXPERIMENTAL) Emit a single sample name from the bam header into an output file.",
+        summary = "If the bam has zero or more than one sample names in the header, this tool will error, by design.\n" +
+                "  This tool has not been tested extensively.  Most options supported by the GATK are irrelevant for this tool.",
+        programGroup = QCProgramGroup.class
+)
+@BetaFeature
+final public class GetSampleName extends GATKTool{
```

Comment:
Why not make this a `CommandLineProgram`?  BTW, if the answer is that emulating `getHeaderForReads` would be too cumbersome, I accept it with no further argument.

Reply:
@davidbenjamin That would be my answer.  I'd be rewriting that code.

Reply:
No action